File: telecomcustomeranalysis/telecom-customer/src/main/scala/com/edu/bigdata/customer/util/PropertiesUtil.scala
package com.edu.bigdata.customer.util

import java.io.InputStreamReader
import java.util.Properties

import com.typesafe.config.ConfigFactory

object PropertiesUtil {

  //第一种配置方式

  /*val config = ConfigFactory.load("kafka.properties")

  def getProperty(key: String): String = {
    config.getString(key)
  }

  def getInt(key: String): Int = {
    config.getInt(key)
  }

  def getBoolean(key: String): Boolean = {
    config.getBoolean(key)
  }

  def getDouble(key: String): Double = {
    config.getDouble(key)
  }

  def getLong(key: String): Long = {
    config.getLong(key)
  }*/

  def getProperties(path:String):Properties={
    val properties:Properties = new Properties()
    properties.load(PropertiesUtil.getClass.getClassLoader.getResourceAsStream(path))
    properties
  }

  val properties:Properties = getProperties("kafka.properties")

  def getProperty(key:String): String ={
      properties.getProperty(key)
  }

  def main(args: Array[String]): Unit = {
    getProperty("bootstrap.servers")
  }

}

File: telecomcustomerserviceanalysis/telecom-customer/src/main/java/com/edu/bigdata/consumer/util/PropertiesUtil.java
package com.edu.bigdata.consumer.util;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

public class PropertiesUtil {

    public static Properties properties = null;

    static {
        // 加载配置文件的属性
        InputStream is = ClassLoader.getSystemResourceAsStream("kafka.properties");
        properties = new Properties();
        try {
            properties.load(is);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static String getProperty(String key) {
        return properties.getProperty(key);
    }
}